### PR TITLE
Zs3server test

### PR DESCRIPTION
**Pull request by shalinikum**

### CODE REVIEW

File: tests/api_tests/zs3server_opertions_test.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		}
+		formData := map[string]string{
+			"file": "@test-file",
+		}
+		resp, err := zs3Client.PutObject(t, queryParams, formData)
+		require.Nil(t, err)
+		require.Equal(t, 500, resp.StatusCode())
+		require.Equal(t, `{"error":"Bucket name contains invalid characters"}`, resp.String())
+	})
+	// t.Run("GetObjects should return 200 all the parameter are correct", func(t *test.SystemTest) {
```
Comment (Kishan-Dhakan) [inline tests/api_tests/zs3server_opertions_test.go:88]: Fix lint issues

Comment (Kishan-Dhakan) [reply tests/api_tests/zs3server_opertions_test.go:88]: Why is GetObjects commented out?

**Review by Kishan-Dhakan — CHANGES_REQUESTED** [review]
remove empty test-file from repo please

File: tests/api_tests/zs3server_opertions_test.go (hunk 1 above)
Comment (shalinikum) [reply tests/api_tests/zs3server_opertions_test.go:88]: Will remove it.

File: tests/api_tests/test-file
Comment (Kishan-Dhakan) [inline tests/api_tests/test-file:1]: Remove this please

**Review by Kishan-Dhakan — COMMENTED** [review]
Happy to merge after passing run

File: tests/api_tests/test-file
Comment (Kishan-Dhakan) [inline tests/api_tests/test-file:1]: Please remove this file

File: tests/api_tests/graph_test.go
Comment (Kishan-Dhakan) [inline tests/api_tests/graph_test.go:1]: Please remove this file

**Review by Kishan-Dhakan — CHANGES_REQUESTED** [review]
Good work, happy to merge post lint fixes and some cleanup

File: tests/api_tests/test-file
Comment (shalinikum) [reply tests/api_tests/test-file:1]: This is importanat file to run the test. As we need to upload a file.

File: tests/api_tests/graph_test.go
Comment (shalinikum) [reply tests/api_tests/graph_test.go:1]: I will remove it.

File: tests/api_tests/zs3server_opertions_test.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
+package api_tests
+
+import (
+	"testing"
+
+	"github.com/0chain/system_test/internal/api/util/test"
+	"github.com/stretchr/testify/require"
+)
+
+// adding comment to run ppipeline again
```
Comment (Kishan-Dhakan) [inline tests/api_tests/zs3server_opertions_test.go:10]: You can simply re-run the previous action, please remove it

File: tests/api_tests/zs3server_opertions_test.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+			"secretAccessKey": SecretAccessKey,
+			"action":          "listObjects",
+			"bucketName":      "system-test",
+		}
+		resp, err := zs3Client.BucketOperation(t, queryParams, map[string]string{})
+		require.Nil(t, err)
+		require.Equal(t, 200, resp.StatusCode())
+	})
+	t.RunSequentially("PutObjects should return 200 all the parameter are correct", func(t *test.SystemTest) {
+		// t.Skip("wait for the issue to get resolved : https://github.com/0chain/zs3server/issues/21")
```
Comment (Kishan-Dhakan) [inline tests/api_tests/zs3server_opertions_test.go:60]: Please remove commented code

File: tests/api_tests/zs3server_opertions_test.go
Comment (Kishan-Dhakan) [inline tests/api_tests/zs3server_opertions_test.go:1]: Filename and test name should have operations, not opertions

File: tests/api_tests/zs3server_opertions_test.go [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+	t.RunSequentially("CreateBucket should return 200 when all the parameters are correct", func(t *test.SystemTest) {
+		queryParams := map[string]string{
+			"accessKey":       AccessKey,
+			"secretAccessKey": SecretAccessKey,
+			"action":          "createBucket",
+			"bucketName":      "system-test",
+		}
+		resp, err := zs3Client.BucketOperation(t, queryParams, map[string]string{})
+		require.Nil(t, err)
+		require.Equal(t, 200, resp.StatusCode())
```
Comment (Kishan-Dhakan) [inline tests/api_tests/zs3server_opertions_test.go:23]: This test already exists in create_bucket_test.go, please remove it if it's exactly same

File: tests/api_tests/zs3server_opertions_test.go [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+	t.RunSequentially("CreateBucket should return error when bucket name already exist", func(t *test.SystemTest) {
+		queryParams := map[string]string{
+			"accessKey":       AccessKey,
+			"secretAccessKey": SecretAccessKey,
+			"action":          "createBucket",
+			"bucketName":      "system-test",
+		}
+		resp, err := zs3Client.BucketOperation(t, queryParams, map[string]string{})
+		require.Nil(t, err)
+		require.Equal(t, 200, resp.StatusCode())
```
Comment (Kishan-Dhakan) [inline tests/api_tests/zs3server_opertions_test.go:34]: If it's an error this should be 4XX or 5XX, please add a TODO comment here and create an issue on zs3server

File: tests/api_tests/zs3server_opertions_test.go (hunk 4 above)
Comment (Kishan-Dhakan) [reply tests/api_tests/zs3server_opertions_test.go:23]: Also take over the content of that file into this one and remove that file

File: tests/api_tests/zs3server_opertions_test.go [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+	})
+
+	t.RunSequentially("ListBucket should return 200 all the parameter are correct", func(t *test.SystemTest) {
```
Comment (Kishan-Dhakan) [inline tests/api_tests/zs3server_opertions_test.go:37]: Already exists in list_buckets_test.go. Please move content from that file to this and remove that file

File: tests/api_tests/zs3server_opertions_test.go [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+			"accessKey":       AccessKey,
+			"secretAccessKey": SecretAccessKey,
+			"action":          "listBuckets",
+		}
+		resp, err := zs3Client.BucketOperation(t, queryParams, map[string]string{})
+		require.Nil(t, err)
+		require.Equal(t, 200, resp.StatusCode())
+	})
+
+	t.RunSequentially("ListObjects should return 200 all the parameter are correct", func(t *test.SystemTest) {
```
Comment (Kishan-Dhakan) [inline tests/api_tests/zs3server_opertions_test.go:48]: Already exists in list_objects_test.go, same as above

File: tests/api_tests/zs3server_opertions_test.go [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+		}
+		formData := map[string]string{
+			"file": "@test-file.txt",
+		}
+		resp, err = zs3Client.PutObject(t, queryParams, formData)
+		require.Nil(t, err)
+		require.Equal(t, 200, resp.StatusCode())
+	})
+	t.RunSequentially("GetObjects should return 200 all the parameter are correct", func(t *test.SystemTest) {
+		// t.Skip("wait for the issue to get resolved : https://github.com/0chain/zs3server/issues/21")
```
Comment (Kishan-Dhakan) [inline tests/api_tests/zs3server_opertions_test.go:85]: Remove comment

File: tests/api_tests/zs3server_opertions_test.go [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+		}
+		formData = map[string]string{
+			"file": "@test-file.txt",
+		}
+		t.Logf(resp.String())
+		require.Nil(t, err)
+		require.Equal(t, 200, resp.StatusCode())
+	})
+
+	t.RunSequentially("PutObjects should return error when buckcet name doesnot exist", func(t *test.SystemTest) {
```
Comment (Kishan-Dhakan) [inline tests/api_tests/zs3server_opertions_test.go:112]: Space between does and not

File: tests/api_tests/zs3server_opertions_test.go [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+			"bucketName":      "system-test",
+		}
+		formData := map[string]string{
+			"file": "@test-file.txt",
+		}
+		resp, err = zs3Client.PutObject(t, queryParams, formData)
+		require.Nil(t, err)
+		require.Equal(t, 200, resp.StatusCode())
+	})
+	t.RunSequentially("GetObjects should return 200 all the parameter are correct", func(t *test.SystemTest) {
```
Comment (Kishan-Dhakan) [inline tests/api_tests/zs3server_opertions_test.go:84]: Please keep a newline between test cases

File: tests/api_tests/zs3server_opertions_test.go [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+			"bucketName":      "system-test",
+			"objectName":      "bucket created as a part of " + t.Name(),
+		}
+		resp, err = zs3Client.BucketOperation(t, queryParams, map[string]string{})
+		require.Nil(t, err)
+		require.Equal(t, 200, resp.StatusCode())
+	})
+
+	t.RunSequentially("RemoveObject should not return error if object doen't exist", func(t *test.SystemTest) {
+		// t.Skip("wait for the issue to get resolved : https://github.com/0chain/zs3server/issues/22")
```
Comment (Kishan-Dhakan) [inline tests/api_tests/zs3server_opertions_test.go:151]: Remove comment

**Review by Kishan-Dhakan — CHANGES_REQUESTED** [review]
Minor changes requested

File: tests/api_tests/zs3server_opertions_test.go (hunk 6 above)
Comment (shalinikum) [reply tests/api_tests/zs3server_opertions_test.go:37]: these two are different

File: tests/api_tests/zs3server_opertions_test.go (hunk 7 above)
Comment (Kishan-Dhakan) [reply tests/api_tests/zs3server_opertions_test.go]: Check this on master https://github.com/0chain/system_test/blob/216311121a1d246a836e907c6f27e327724bd95b/tests/api_tests/list_objects_test.go#L12, it already exists.

File: tests/api_tests/zs3server_opertions_test.go (hunk 4 above)
Comment (Kishan-Dhakan) [reply tests/api_tests/zs3server_opertions_test.go]: This is not resolved

**Review by Kishan-Dhakan — APPROVED** [review]
LGTM

**Pull request merged**